src/snoowrap.js

```diff
@@ -639,6 +639,45 @@ const snoowrap = class snoowrap {
     return this._get({uri: 'api/saved_categories'}).get('categories');
   }
 
+  /**
+   * @param {string} query a query to search
+   * @param {Object} [config] config object
+   * @returns {Promise} A promise with array os subreddits starting with that query
+   * @example
+   * const config = {
+   *  exact:false,
+   *  include_over_18:false,
+   *  include_unadvertisable:false
+   * };
+   * const {subreddits} = await r.searchSubreddit("learn", config);
+   * => [
+   * {
+   *   active_user_count: 1705
+   *   allow_chat_post_creation: null
+   *   allow_images: false
+   *   icon_img: ""
+   *   key_color: ""
+   *   name: "learnprogramming"
+   *   subscriber_count: 1084731
+   * },
+   * {
+   *   active_user_count: 126
+   *   allow_chat_post_creation: null
+   *   allow_images: true
+   *   icon_img: ""
+   *   key_color: ""
+   *   name: "learnjavascript"
+   *   subscriber_count: 76606
+   * }
+   * ]
+   */
+  searchSubreddit (query, config = {exact: false, include_over_18: false, include_unadvertisable: false}) {
```

Comment:
I'd consolidate this to this, though update docs to have options instead of config:
```suggestion
  searchSubreddit (options = {
  	query: requiredArg('query'), 
  	exact: false, 
  	include_over_18: false, 
  	include_unadvertisable: false
 }) {
```